### ISSUE TITLE
Log how long the Scheduler queue gets

### ISSUE THREAD

**Issue by Kegsay** [matrix-org/matrix-appservice-irc#188]
Over the weekend, Freenode had a huge netsplit. `Scheduler` logic correctly kicked in, making a humongous queue for reconnections. It looks like this queue was taking forever to clear (probably due to said netsplit). Some numbers: Between 18:00~19:00 on 2016-09-24, there were 1724 reconnections queued up, but only 74 successful connections (that is, the connection got `rpl_welcome`). This led to massive head-of-line blocking, effectively stopping M->I traffic

**Comment by Kegsay** [comment]
We're kinda stuck between a rock and a hard place with this. By queueing sensibly, we avoid being k-lined from reconnect speed, but it means that things like this can happen.

**Comment by Kegsay** [comment]
Here's a fun graph illustrating the severity of that netsplit: (Red is Freenode)
![screencapture-matrix-org-prometheus-graph-1474880459627](https://cloud.githubusercontent.com/assets/7190048/18828552/4e04bf98-83d0-11e6-8532-321bb118087e.png)

When the red line hits 0 that is when it was restarted.

**Comment by Kegsay** [comment]
@leonerd suggests that we can mitigate this a little by effectively allowing a certain number of "simultaneous connections" - that is, we allow N connections to be up-in-the-air at any one time. This helps against 1 nasty TCP connection slowing everything down. We would need to experiment with what to set N to.

Of course, the long term solution is to just use IPv6....

**Comment by Kegsay** [comment]
@ara4n - thoughts?

**Comment by Kegsay** [comment]
tl;dr - Use IPv6 on Freenode.

**Comment by Kegsay** [comment]
- Tagged as support to move Freenode to IPv6 (we need to speak to them first).
- Tagged as bug because we really do need that logging in general.

**Comment by lukebarnard1** [comment]
Freenode bridge is now using IPv6 and scheduled queue length is being logged.
